### PR TITLE
Add timeline visualization to query info ui

### PR DESCRIPTION
**Pull request by haozhun**
e.g. screenshot showing hash skew:

screen shot 2015-04-30 at 4 20 31 pm

### CODE REVIEW

**Comment by haozhun** [comment]
Moved visualization to a separate page as suggested by @martint

File: presto-main/src/main/resources/webapp/timeline.html [hunk 1]
```diff
@@ -0,0 +1,132 @@
+<!DOCTYPE html>
+<head xmlns="http://www.w3.org/1999/html" xmlns="http://www.w3.org/1999/html">
```
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: I think it's recommended to have an `<html>`

File: presto-main/src/main/resources/webapp/timeline.html [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    <script src="vendor/d3/d3-3.3.4.js"></script>
+
+    <link href="vendor/bootstrap/css/bootstrap.css" rel="stylesheet" type="text/css">
+
+    <link rel="stylesheet" href="vendor/vis/dist/vis.min.css">
+    <script src="vendor/vis/dist/vis.min.js"></script>
+
+    <style type="text/css">
+        .vis.timeline .item {
+        padding: 0px;
```
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: indent all the properties to match the style of `index.html`

File: presto-main/src/main/resources/webapp/timeline.html [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        }
+
+        .vis.timeline .labelset .vlabel .inner {
+        padding: 0px;
```
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: these too

File: presto-main/src/main/resources/webapp/timeline.html [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+        }
+
+        .vis.timeline .item.range {
+        height: 3px;
```
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: and these, and the rest of them

File: presto-main/src/main/resources/webapp/timeline.html [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+            },
+        };
+    });
+
+    var groups = new vis.DataSet();
+    var items = new vis.DataSet();
+    for (var i = 0; i < tasks.length; i++) {
+        var task = tasks[i];
+        var taskIdMain = task.taskId.substr(0, task.taskId.indexOf("."));
+        var taskIdMinor = task.taskId.substr(task.taskId.indexOf(".")+1);
```
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: spaces around `+`

File: presto-main/src/main/resources/webapp/timeline.html [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+                end: task.stats.endTime,
+            },
+        };
+    });
+
+    var groups = new vis.DataSet();
+    var items = new vis.DataSet();
+    for (var i = 0; i < tasks.length; i++) {
+        var task = tasks[i];
+        var taskIdMain = task.taskId.substr(0, task.taskId.indexOf("."));
```
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: name this `stageId`

File: presto-main/src/main/resources/webapp/timeline.html (hunk 5 above)
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: and this `taskId`

File: presto-main/src/main/resources/webapp/timeline.html [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+        var taskIdMinor = task.taskId.substr(task.taskId.indexOf(".")+1);
+        if (taskIdMinor == 0) {
+            groups.add({
+                id: taskIdMain,
+                content: task.taskIdMain,
```
Comment (cberner) [inline presto-main/src/main/resources/webapp/timeline.html]: where was `task.taskIdMain` set? it looks like this is going to evaluate to `undefined`

**Comment by cberner** [comment]
Had a question and a couple comments. Assign back to me after you're done

**Comment by haozhun** [comment]
Also added legend. Ready for review.

screen shot 2015-05-13 at 2 34 41 pm

**Comment by cberner** [comment]
looks good

**Comment by electrum** [comment]
This looks great!

**Pull request merged**